src/ibtracert.c

```diff
@@ -811,9 +820,29 @@ int main(int argc, char **argv)
 	argc -= optind;
 	argv += optind;
 
-	if (argc < 2)
+	if (argc < 2 && lid_file[0] == '\0') 
 		ibdiag_show_usage();
 
+	if (argc == 2) {
+		srcid = strdup(argv[0]);
+		dstid = strdup(argv[1]);
+     	}
+
+	if (lid_file != NULL) {
+		lid_fd = fopen(lid_file, "r");
+		if (!lid_fd) {
+			IBEXIT("cannot open lid-file %s", lid_file);
+		} else {
+			srcid = (char*) malloc(6*sizeof(char));
+			dstid = (char*) malloc(6*sizeof(char));
+			lidfd_res = fscanf(lid_fd, "%s %s", srcid, dstid);
+			if (lidfd_res != 2) {
+				IBEXIT("lid-file empty or contains bad data, %i, %s",lidfd_res,lid_file);
```

Comment:
lidfd_res could be EOF
Would this (when != 2) be better as:
if (lid_res != EOF)
IBEXIT("lid-file (%s) contains bad data %i", lid_file, lidfd_res);
else
IBEXIT("lid-file (%s) empty %s", lid_file, strerror(errno));

Reply:
hrm - != 2 seems to cover all bases, whether a matching failure or empty/bad file.  lidfs_res of 2 (meaning I got my 2 values) would != EOF, but that is not an error.  Am I missing something here ?

Reply:
It was just an attempt to improve the error message when there wasn't 2 arguments.fscanf says when EOF is returned that errno contains the error so I was trying to separate out that case from the other case.  It's probably overkill.

Reply:
Yah - I started out with 2 different error messages, but collapsed them cuz it didn't seem necessary. ...I think I'll leave it as is ...